src/services/services.ts

```diff
@@ -7310,6 +7312,40 @@ namespace ts {
             return { newText: result, caretOffset: preamble.length };
         }
 
+        function isValidBraceCompletionAtPostion(fileName: string, position: number, openingBrace: number): boolean {
+
+            // '<' is currently not supported, figuring out if we're in a Generic Type vs. a comparison is too 
+            // expensive to do during typing scenarios
+            // i.e. whether we're dealing with:
+            //      var x = new foo<| ( with class foo<T>{} )
+            // or 
+            //      var y = 3 <|

```

Comment:
Why not always complete, even in the `var y = 3 < {` case? If it's just a typo, then the user will just backspace immediately, which should delete both the `{` and the `}`.